core/src/apps/bitcoin/sign_tx/bitcoin.py

```diff
@@ -76,6 +77,10 @@ async def signer(self) -> None:
             self.orig_txs,
         )
 
+        # Following steps can take a long time, make sure autolock doesn't kick in.
+        # This is set to True again after workflow is finished in start_default().
+        workflow.autolock_interrupts_workflow = False
+
```

Comment:
So now if a user leaves the Trezor hanging in an output confirmation screen, leaves the room and an attacker comes in, the attacker can finish the transaction confirmation even if the device has been autolocked. In comparison if the user leaves the Trezor hanging in a change label screen, i.e. a far less security-sensitive operation, the device will stop the workflow after the autolock timeout and won't let a potential attacker finish the label change.

Reply:
Oh yeah, you are absolutely right. As usual I am doing two things at once and I overlooked that steps 1, 2 & approval are already complete at this stage.

> only if a prompt pops up afterwards, then it would remain open. (something something late path warning? or did we already resolve that?)

That would be a problem, but I think that's already resolved. All checks and warnings should pop up in Steps 1, 2 and the approval step. After that if some check fails, then the entire signing process should fail.

Reply:
For example `BasicApprover.add_internal_input()` shows a warning but `BasicApprover.check_internal_input()` raises an exception. So I think we are fine.